solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -105,7 +107,17 @@ public int hashCode() {
     result = (prime * result) + ((meta == null) ? 0 : meta.hashCode());
     result = (prime * result)
         + ((originalQuery == null) ? 0 : originalQuery.hashCode());
-    result = (prime * result) + ((efi == null) ? 0 : efi.hashCode());
+    if (efi == null) {
+      result = (prime * result) + 0;
+    }
+    else {
+      for (final Entry<String,String[]> entry : efi.entrySet()) {
```

Comment:
necessary?  
old code used to do `result = (prime * result) + ((efi == null) ? 0 : efi.hashCode());`